src/applications/vaos/covid-19-vaccine/redux/actions.js

```diff
@@ -293,38 +311,37 @@ export function getAppointmentSlots(startDate, endDate, initialFetch = false) {
       dispatch({ type: FORM_CALENDAR_FETCH_SLOTS });
 
       try {
-        const startDateString = !fetchedStartMonth
-          ? format(new Date(startDate), 'yyyy-MM-dd')
-          : format(startOfMonth(new Date(endDate)), 'yyyy-MM-dd');
-        const endDateString = !fetchedEndMonth
-          ? format(new Date(endDate), 'yyyy-MM-dd')
-          : format(endOfMonth(new Date(startDate)), 'yyyy-MM-dd');
-
         const fetchedSlots = await getSlots({
           siteId,
           clinicId: data.clinicId,
-          startDate: startDateString,
-          endDate: endDateString,
+          startDate: startOfMonth(startDate),
+          endDate: endOfMonth(endDate),
           convertToUtc: featureConvertSlotsToUTC,
         });
 
         if (initialFetch) {
           recordItemsRetrieved('covid_slots', fetchedSlots?.length);
         }
 
-        const now = new Date();
-        mappedSlots = fetchedSlots.filter(slot =>
-          isAfter(new Date(slot.start), now),
-        );
+        mappedSlots = fetchedSlots.filter(slot => {
+          return isAfter(
+            new Date(slot.start),
+            new Date(new Date().toISOString()),
+          );
+        });
```

Comment:
This is important. When comparing dates using isAfter or isBefore, both dates need to be created in the same format. In this case UTC.